bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/parser/AbstractBaseParser.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.stiebeleltron.internal.parser;
+
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.openhab.io.transport.modbus.ModbusBitUtilities;
+import org.openhab.io.transport.modbus.ModbusConstants.ValueType;
+import org.openhab.io.transport.modbus.ModbusRegisterArray;
+
+/**
+ * Base class for parsers with some helper methods
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ * @author Paul Frank - Added more methods
+ */
+@NonNullByDefault
+public class AbstractBaseParser {
+
+    /**
+     * Extract an optional double value
+     *
+     * @param raw the register array to extract from
+     * @param index the address of the field
+     * @return the parsed value or empty if the field is not implemented
+     */
+    protected Optional<Double> extractOptionalDouble(ModbusRegisterArray raw, int index) {
+        return ModbusBitUtilities.extractStateFromRegisters(raw, index, ValueType.INT16)
+                .map(value -> ((float) value.intValue()) / 10.0).filter(value -> value != (short) 0x8000);
+    }
```

Comment:
Since this method returns Double, you might want to use `double` instead of `float`.

Reply:
changed to double